### PR TITLE
Compatibility with PHP 7.2

### PR DESCRIPTION
**Pull request by aleswita**
- remove dependency on Nette\Utils\Object
- add methods param and return type hints
- strict types
- Nette Coding Standard

**BC breaks**
- require PHP ^7.1
- require Nette 2.4 || 3.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,10 +11,11 @@
...
 	"require": {
-		"nette/application": "~2.2@dev",
-		"nette/di": "~2.2@dev",
-		"nette/http": "~2.2@dev",
-		"nette/utils": "~2.2@dev"
+		"php": "^7.1",
+		"nette/application": "~2.4.0 || ~3.0.0",
+		"nette/di": "~2.4.0 || ~3.0.0",
+		"nette/http": "~2.4.0 || ~3.0.0",
+		"nette/utils": "~2.4.0 || ~3.0.0"
```
Comment (enumag) [inline composer.json]: Use `^` instead of `~`.

**Review by enumag — CHANGES_REQUESTED** [review]
The diff is all screwed up somehow. I'm guessing you accidently changed line endings or indentation. Fix that so that I can review your changes.

I mean I can review them with ?w=1 but then I can't comment on them...

Also use https://github.com/Kdyby/StrictObjects instead of Nette\SmartObject.

**Comment by aleswita** [comment]
@enumag I'm working on Windows, so I was change the newlines to Unix format, now difs looks good

**Comment by enumag** [comment]
Add the kdyby strict-objects dependency to composer.json. Otherwise it looks good.

**Comment by aleswita** [comment]
@enumag done

**Comment by enumag** [comment]
Also I'm thinking about merging https://github.com/Kdyby/Wkhtmltopdf/pull/6 first since it has travis setup and other good things. Could you review that PR and comment there what do you think about it?

**Comment by aleswita** [comment]
Seems good, so if PR#6 are ok, merge first and I will try resolved conflicts.

**Comment by enumag** [comment]
#6 is merged now. Also if you could do the change suggested [here](https://github.com/Kdyby/Wkhtmltopdf/pull/6/files#r30045182) it would be great.

**Comment by aleswita** [comment]
@enumag one question, I have no experience with **Kdyby/StrictObjects**, its posible to read private variable with public getter like as **Nette\SmartObject**?

**Comment by enumag** [comment]
@aleswita I have no idea. Personally I never used neither Kdyby/StrictObjects nor Nette\SmartObject. I'd be ok with not using any such trait in Kdyby packages but it's probably better to keep it consistent.

**Comment by aleswita** [comment]
@enumag seems done

**Comment by dada-amater** [comment]
Will this be merged? It is our php7.2 upgrade blocker. Thank you.

**Comment by enumag** [comment]
I was focusing on Kdyby/Doctrine so I kind of forgot about this one. I need to review it but yes I intend to merge it soon. Thanks for reminding me.

**Review by enumag — CHANGES_REQUESTED** [review]
Remove useless annotations like `@return void` and other cases where `@param` and `@return` just duplicate the method signature.

**Comment by dada-amater** [comment]
Any update? And please make new release, in composer there is still old v0.1.

**Comment by solcik** [comment]
@enumag Could you merge this one and I will remove useless annotations in another PR as you requested above?

**Comment by enumag** [comment]
@solcik Ok. Thinking about it more I'm not completely sure if the annotations aren't required by the code checker. Better test that before you remove them everywhere.

**Comment by dada-amater** [comment]
@enumag @solcik @aleswita Thank you!

**Pull request merged**